core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1882,14 +1882,15 @@ private void executeCopy0(SqlExecutionContext executionContext, CopyModel model)
             int workerCount = executionContext.getWorkerCount();
             ExpressionNode fileNameNode = model.getFileName();
             final CharSequence fileName = fileNameNode != null ? GenericLexer.assertNoDots(GenericLexer.unquote(fileNameNode.token), fileNameNode.position) : null;
-            if (model.isParallel()) {
                 if (workerCount < 1) {
                     throw SqlException.$(0, "Invalid worker count set [value=").put(workerCount).put("]");
                 }
-
                 if (model.isCancel()) {
                     addTextImportRequest(model, null);
                 } else {
+                    if (model.getTimestampColumnName() == null && (model.getPartitionBy() != -1 || model.getTimestampFormat() != null)) {
```

Comment:
We seem to support PartitionBy.NONE + timestamp in serial importer (but not in the parallel one), so we should probably relax this check.